tests/integration/resource_aware_pool_tests.cpp

```diff
@@ -0,0 +1,156 @@
+/////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-24, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include <stdio.h>
+
+#include <iostream>
+
+#include "camp/camp.hpp"
+#include "gtest/gtest.h"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/Umpire.hpp"
+#include "umpire/config.hpp"
+#include "umpire/strategy/ResourceAwarePool.hpp"
+
+using namespace camp::resources;
+
+void host_sleep(int* ptr)
+{
+  int i = 0;
+  while (i < 1000000) {
+    int y = i;
+    y++;
+    i = y;
+  }
+  *ptr = i;
+  ptr++;
+}
+
+TEST(ResourceAwarePool_Host_Test, Check_States_Host)
+{
+  auto& rm = umpire::ResourceManager::getInstance();
+  auto pool = rm.makeAllocator<umpire::strategy::ResourceAwarePool>("rap-pool-host", rm.getAllocator("HOST"));
+
+  Resource r1{Host{}}, r2{Host{}};
+  int* ptr = static_cast<int*>(pool.allocate(r1, 1024));
+  int* compare_ptr1 = ptr;
+
+  EXPECT_EQ(getResource(pool, ptr), r1);
+  EXPECT_EQ(getPendingSize(pool), 0);
+
+  host_sleep(ptr);
```

Comment:
I don't think this sleep is necessary here.

Reply:
Or is it for consistency with other tests?

Reply:
Yes, it is to show that even though I am creating two different host resources, they are really the same resource when you boil it down.. so even though there is a sleep in between allocating and reallocating, it doesn't matter, a data race isn't possible.. I added some docs to explain that, but haven't committed that yet